lib/hotel_controller.rb

```diff
@@ -1,24 +1,44 @@
 module Hotel
-  class HotelController
-    # Wave 1
-    def rooms
-      # You might want to replace this method with an attr_reader
-      return []
-    end
+  class HotelController 
 
-    def reserve_room(start_date, end_date)
-      # start_date and end_date should be instances of class Date
-      return Reservation.new(start_date, end_date, nil)
-    end
+    attr_accessor :hotel_rooms, :reservations
+    attr_reader 
 
-    def reservations(date)
-      return []
+    def initialize()
+      @hotel_rooms = (1..20).to_a
+      @reservations = []
     end
 
-    # Wave 2
-    def available_rooms(start_date, end_date)
-      # start_date and end_date should be instances of class Date
-      return []
+    # Wave 1
+    # creates a reservation of a room for a given date range
+    def reserve_room(start_date, end_date, room_number, rate)
```

Comment:
Nice work implementing this complex logic. To fully tie together `reserve_room` and `available_rooms`, ideally `reserve_room` would not take a `room_number` as an argument, but instead find an available room by using the `available_rooms` method.